sonata/main.py

```diff
@@ -2515,9 +2515,17 @@ def mpd_stop(self, _widget, _key=None):
             self.iterate_now()
 
     def mpd_prev(self, _widget, _key=None):
-        if self.conn:
-            self.mpd.previous()
-            self.iterate_now()
+	if self.conn:
+            if self.status_is_play_or_pause():
+                at, length = [int(c) for c in self.status['time'].split(':')]
+                if at >= consts.PREV_TRACK_RESTART:
+                    self.seek(int(self.status['song']), 0)
+                else:
+                    self.mpd.previous()
```

Comment:
I know the original code isn't exactly high quality, but to remove a bit of repetition, you can return directly in the `if at >= consts.PREV_TRACK_RESTART` body, remove the 2x `else` in here and have the single call to `.previous()`. Something like this:

```python
    if self.conn:
            if self.status_is_play_or_pause():
                # Try to rewind the song if we are advanced enough...
                at, length = [int(c) for c in self.status['time'].split(':')]
                if at >= consts.PREV_TRACK_RESTART:
                    self.seek(int(self.status['song']), 0)
                    return

            # otherwise, just go to the previous song
            self.iterate_now()
            self.mpd.previous()
```

Reply:
Yeah, that's much better. Thanks!